src/components/Navigator/NavigatorCard.vue

```diff
@@ -1014,37 +1018,65 @@ export default {
         if (lastActivePathItem === currentActiveItem.path) {
           return;
         }
-        // Get the surrounding items
-        const siblings = getSiblings(this.activeUID, this.childrenMap, this.children);
-        const children = getChildren(this.activeUID, this.childrenMap, this.children);
-        const parents = getParents(this.activeUID, this.childrenMap);
-        // try to match if any of the `siblings`,`children` or any of the `parents`,
-        // match the current open item
-        const matchingItem = [...children, ...siblings, ...parents]
-          .find(child => child.path === lastActivePathItem);
+        // try to match current open item in its surroundings, starting with active item
+        if (this.matchSurroundingItems(this.activeUID, lastActivePathItem)) return;
 
-        // set the match as an active item
-        if (matchingItem) {
-          this.setActiveUID(matchingItem.uid);
-          return;
+        if (this.hasValidHash) {
+          // if no match, try again to match with generic item
+          // Needed for continuing to highlight current generic page
+          // when selecting an overload from dropdown that's also specifically curated in elsewhere
+          const genericItem = lastActivePathItem.split('-')[0];
```

Comment:
If by any chance we have path like `Foo-bar-f1234` this will use `Foo` which is wrong.